paasta_tools/check_flink_services_health.py

```diff
@@ -105,6 +108,23 @@ def check_under_registered_taskmanagers(
     return (unhealthy, output)
 
 
+def get_cr_name(si_pods: Sequence[V1Pod]) -> str:
+    """Returns the flink custom resource name based on the pod name.  We are randomly choosing jobmanager pod here.
+    This change is related to FLINK-3129
+    """
+    jobmanager_pod = [
+        pod
+        for pod in si_pods
+        if pod.metadata.labels["flink.yelp.com/container-type"] == "jobmanager"
+        and is_pod_ready(pod)
+        and container_lifetime(pod).total_seconds() > 60
+    ]
+    if len(jobmanager_pod) == 1:
+        return jobmanager_pod[0].metadata.name.split("-jobmanager-")[0]
+    else:
+        return ""
```

Comment:
what do you think about erroring out here rather than returning an empty string?

Reply:
I handled this case in `check_under_registered_taskmanagers`.

Reply:
Yeah I see it's handled there, I just wonder whether a more specific error message here is useful. Not a big deal though, don't block on this.

Reply:
Yeah, my understanding is if the jobmanager is not running we won't have running dashboard which will be pointed out by the output message handling. Also, in such cases jobmanager pod check will also fail.